tools/buildsys/src/manifest.rs

```diff
@@ -33,6 +33,35 @@ url = "https://bar"
 sha512 = "123456"
 ```
 
+The `bundle-*` keys on `external-files` are a group of optional modifiers
+and are used to untar an upstream external file archive,
+vendor any dependent code, and bundle it back up to be consumed.
+All three options are required when bundling an archive's dependences.
+
+`bundle-modules` is a list of module "paradigms" the external-file should
+be vendored through. For example, if a project contains a go.mod and go.sum file,
+"go" may be entered to vendor the dependencies.
+Currently, only "go" is supported.
+
+`bundle-path` is the filepath _within_ the archive that contains the module.
+So, for example, given a go project that has the necessary go.mod and go.sum
+files within the archive located at the filepath a/b/c,
+this `bundle-path` value should be `a/b/c`.
+
+`bundle-output-path` is the path of the output tar archive.
+This name may be referenced within a spec, when creating a package, etc.
+For example, `my-package-bundled.tar.gz` may be entered as an output path
+indicating the package is "bundled" and has been archived.
```

Comment:
A default value using a convention where `-bundled` is appended to the last part of the external filename before `.tar.*` would be nice.

Reply:
👍🏼 I punted on this _abit_ since I was having trouble with Rust string manipulation. It now defaults to `bundled-{name-of-tar}`. So, for example, in the `hotdog` package, the default would be `bundled-hotdog-b85b755.tar.gz` if `bundled-output-path` is not provided.